assets/src/settings-page/analytics.js

```diff
@@ -0,0 +1,307 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useContext, useEffect, useRef, useState, useCallback, useMemo } from '@wordpress/element';
+import { Icon, plus, trash } from '@wordpress/icons';
+import { Button, TextControl, PanelRow, BaseControl } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { AMPDrawer } from '../components/amp-drawer';
+import { Options } from '../components/options-context-provider';
+
+/**
+ * Prefix for the temporary ID assigned to new entries. Entries receive their permanent ID when saved to the database.
+ */
+const NEW_ENTRY_KEY_PREFIX = '__new__';
+
+/**
+ * Provides a unique temporary key for new entries. Will be replaced with a key generated on the backend when settings are saved.
+ *
+ * @param {string} prefix The prefix for new keys.
+ */
+function useUniqueNewKey( prefix = NEW_ENTRY_KEY_PREFIX ) {
+	const [ currentIndex, setCurrentIndex ] = useState( 0 );
+
+	const getNewKey = useCallback( () => {
+		setCurrentIndex( ( oldIndex ) => oldIndex + 1 );
+		return `${ prefix }-${ currentIndex }`;
+	}, [ currentIndex, prefix ] );
+
+	return getNewKey;
+}
+
+/**
+ * Component for a single analytics entry.
+ *
+ * @param {Object} props Component props.
+ * @param {string} props.entryId Unique ID for the entry.
+ * @param {Function} props.onChange Callback to run when data changes.
+ * @param {Function} props.onDelete Callback to run when the entry is to be DocumentAndElementEventHandlers.
+ * @param {string} props.type The entry type.
+ * @param {string} props.config The config JSON string.
+ */
+function AnalyticsEntry( { entryId = '', onChange, onDelete, type = '', config = '{}' } ) {
+	const isExistingEntry = 0 !== entryId.indexOf( NEW_ENTRY_KEY_PREFIX );
+
+	const analyticsTitle = useMemo( () => {
+		if ( isExistingEntry ) {
+			const newEntrySlug = sprintf(
+				'%1$s%2$s',
+				type ? type + '-' : '',
+				entryId.substr( entryId.length - 6 ),
+			);
```

Comment:
I template literal would do here, no?

```suggestion
			const newEntrySlug = `${ type ? type + '-' : '' }${ entryId.substr( entryId.length - 6 ) }`;
```